pkg/operator/controller/controller.go

```diff
@@ -399,9 +399,41 @@ func (r *reconciler) getClusterIPFromNetworkConfig() (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("invalid service cidr %v: %v", serviceCIDR, err)
 	}
+	// Ensure dnsClusterIP is not allocated to an existing service.
+	if err := r.ensureServiceIP(dnsClusterIP); err != nil {
+		return "", fmt.Errorf("failed to ensure service IP %s: %v", dnsClusterIP, err)
```

Comment:
Returning an error here causes `ensureDNS` to return before it updates the DNS's status.  It might be better to perform this check in `ensureDNSService`.  (It might also make sense for `ensureDNS` to update the DNS's status even if `getClusterIPFromNetworkConfig` returns an error, but maybe as a separate change.)

Reply:
@Miciah thanks for the review! Commit `4ce15d5` addresses the feedback above. `ensureDNSService()` will now return an error if the DNS Service IP is allocated to another Service. Since `ensureDNS()` is aggregating the errors, `syncDNSStatus()` should still get called to update status.